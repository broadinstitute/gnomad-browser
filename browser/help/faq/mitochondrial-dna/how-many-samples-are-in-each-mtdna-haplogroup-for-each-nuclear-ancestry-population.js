import React, { useEffect, useState } from 'react'
import styled from 'styled-components'

import { BaseTable, Select } from '@gnomad/ui'

import rawHaplogroupAndAncestryData from './how-many-samples-are-in-each-mtdna-haplogroup-for-each-nuclear-ancestry-population.json'

const haplogroupAndAncestryData = rawHaplogroupAndAncestryData.data.map((row) => {
  const [haplogroup, ancestry, n] = row.split(' ')
  return { haplogroup, ancestry, n }
})

const codeToHaplogroupName = {
  afr: 'African/African American',
  ami: 'Amish',
  amr: 'Latino/Admixed American',
  asj: 'Ashkenazi Jewish',
  eas: 'East Asian',
  fin: 'European (Finnish)',
  mid: 'Middle Eastern',
  nfe: 'European (non-Finnish)',
  oth: 'Other',
  sas: 'South Asian',
}

const HaplogroupAndAncestryBaseTable = styled(BaseTable)`
  margin-top: 1em;
  margin-bottom: 1em;

  td {
    text-align: right;
  }
`

const HaplogroupOrAncestrySelector = styled(Select)`
  margin: 1em 1em 0 0;
`

const HaplogroupAndAncestryFilterTable = () => {
  const [haplogroupSelected, setHaplogroupSelected] = useState('All')
  const [ancestrySelected, setAncestrySelected] = useState('All')
  const [filteredData, setFilteredData] = useState([])
  const haplogroupOptions = [
    'A',
    'B',
    'C',
    'D',
    'E',
    'F',
    'G',
    'H',
    'HV',
    'I',
    'J',
    'K',
    'L0',
    'L1',
    'L2',
    'L3',
    'L4',
    'L5',
    'M',
    'N',
    'P',
    'R',
    'T',
    'U',
    'V',
    'W',
    'X',
    'Y',
    'Z',
  ]
  const ancestryOptions = [
    'African/African American',
    'Amish',
    'Latino/Admixed American',
    'Ashkenazi Jewish',
    'East Asian',
    'European (Finnish)',
    'Middle Eastern',
    'European (non-Finnish)',
    'South Asian',
    'Other',
  ]

  useEffect(() => {
    const newFilteredData = haplogroupAndAncestryData.filter((row) => {
      return (
        (row.haplogroup === haplogroupSelected || haplogroupSelected === 'All') &&
        (codeToHaplogroupName[row.ancestry] === ancestrySelected || ancestrySelected === 'All')
      )
    })
    setFilteredData(newFilteredData)
  }, [haplogroupSelected, ancestrySelected])

  return (
    <>
      <HaplogroupOrAncestrySelector
        id="haplogroup-selected"
        value={haplogroupSelected}
        onChange={(e) => {
          setHaplogroupSelected(e.target.value)
        }}
      >
        <optgroup label="Haplogroup">
          <option value="All" key="All">
            All Haplogroups
          </option>
          {haplogroupOptions.map((haplogroup) => (
            <option value={haplogroup} key={haplogroup}>
              {haplogroup}
            </option>
          ))}
        </optgroup>
      </HaplogroupOrAncestrySelector>

      <HaplogroupOrAncestrySelector
        id="ancestry-selected"
        value={ancestrySelected}
        onChange={(e) => {
          setAncestrySelected(e.target.value)
        }}
      >
        <optgroup label="Ancestry">
          <option value="All" key="All">
            All Ancestries
          </option>
          {ancestryOptions.map((ancestry) => (
            <option value={ancestry} key={ancestry}>
              {ancestry}
            </option>
          ))}
        </optgroup>
      </HaplogroupOrAncestrySelector>

      <HaplogroupAndAncestryBaseTable>
        <thead>
          <tr>
            <th scope="col">Haplogroup</th>
            <th scope="col">Population</th>
            <th scope="col">Number of Samples</th>
          </tr>
        </thead>

        <tbody>
          {filteredData.map((row) => {
            return (
              <tr key={row.haplogroup + row.ancestry}>
                <th scope="row">{row.haplogroup}</th>
                <th>{codeToHaplogroupName[row.ancestry]}</th>
                <td>{row.n}</td>
              </tr>
            )
          })}
        </tbody>

        <tfoot>
          <tr>
            <th scope="row">Haplogroup: {haplogroupSelected}</th>
            <th>Ancestry: {ancestrySelected}</th>
            <td>
              Samples:{' '}
              {filteredData.reduce((acc, row) => {
                return acc + parseInt(row.n, 10)
              }, 0)}
            </td>
          </tr>
        </tfoot>
      </HaplogroupAndAncestryBaseTable>
    </>
  )
}

export const question =
  'How many samples are in each mtDNA haplogroup, for each nuclear ancestry population?'

export const renderAnswer = () => <HaplogroupAndAncestryFilterTable />
