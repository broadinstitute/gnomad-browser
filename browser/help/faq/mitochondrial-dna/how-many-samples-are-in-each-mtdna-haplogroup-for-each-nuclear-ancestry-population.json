{
  "schema": "hap pop n",
  "data": [
    "A afr 94",
    "A amr 2380",
    "A eas 74",
    "A nfe 43",
    "A oth 84",
    "A sas 5",
    "B afr 95",
    "B amr 1178",
    "B eas 208",
    "B nfe 22",
    "B oth 32",
    "B sas 2",
    "C afr 53",
    "C amr 687",
    "C eas 47",
    "C nfe 37",
    "C oth 39",
    "C sas 5",
    "D afr 37",
    "D amr 182",
    "D eas 309",
    "D fin 12",
    "D nfe 29",
    "D oth 12",
    "D sas 22",
    "E afr 11",
    "E amr 2",
    "E eas 11",
    "E nfe 4",
    "E oth 6",
    "F afr 21",
    "F amr 3",
    "F eas 229",
    "F nfe 6",
    "F sas 23",
    "G afr 1",
    "G eas 60",
    "G fin 5",
    "G nfe 17",
    "G sas 8",
    "H afr 306",
    "H ami 164",
    "H amr 329",
    "H asj 357",
    "H eas 1",
    "H fin 2198",
    "H mid 5",
    "H nfe 11226",
    "H oth 152",
    "H sas 46",
    "HV afr 9",
    "HV amr 10",
    "HV asj 68",
    "HV fin 18",
    "HV mid 2",
    "HV nfe 546",
    "HV oth 6",
    "HV sas 42",
    "I afr 25",
    "I amr 5",
    "I asj 30",
    "I eas 1",
    "I fin 114",
    "I nfe 743",
    "I oth 8",
    "I sas 8",
    "J afr 61",
    "J amr 141",
    "J asj 107",
    "J fin 245",
    "J mid 3",
    "J nfe 2520",
    "J oth 41",
    "J sas 26",
    "K afr 52",
    "K amr 30",
    "K asj 460",
    "K fin 194",
    "K mid 1",
    "K nfe 1952",
    "K oth 33",
    "K sas 10",
    "L0 afr 605",
    "L0 amr 35",
    "L0 nfe 10",
    "L0 oth 13",
    "L1 afr 2760",
    "L1 amr 149",
    "L1 fin 1",
    "L1 nfe 28",
    "L1 oth 38",
    "L1 sas 1",
    "L2 afr 4404",
    "L2 amr 160",
    "L2 asj 34",
    "L2 eas 1",
    "L2 mid 1",
    "L2 nfe 44",
    "L2 oth 78",
    "L2 sas 2",
    "L3 afr 5282",
    "L3 amr 242",
    "L3 asj 1",
    "L3 eas 1",
    "L3 nfe 39",
    "L3 oth 104",
    "L3 sas 3",
    "L4 afr 118",
    "L4 amr 4",
    "L4 nfe 2",
    "L4 oth 2",
    "L5 oth 1",
    "M afr 113",
    "M amr 3",
    "M asj 10",
    "M eas 341",
    "M fin 1",
    "M nfe 64",
    "M oth 3",
    "M sas 763",
    "N afr 2",
    "N amr 1",
    "N asj 122",
    "N eas 52",
    "N fin 7",
    "N mid 1",
    "N nfe 161",
    "N oth 9",
    "N sas 11",
    "P eas 5",
    "P nfe 2",
    "R afr 5",
    "R amr 4",
    "R asj 28",
    "R eas 103",
    "R fin 29",
    "R mid 2",
    "R nfe 69",
    "R oth 4",
    "R sas 149",
    "T afr 66",
    "T ami 13",
    "T amr 34",
    "T asj 63",
    "T fin 239",
    "T nfe 2599",
    "T oth 34",
    "T sas 32",
    "U afr 185",
    "U ami 190",
    "U amr 101",
    "U asj 56",
    "U fin 1238",
    "U mid 3",
    "U nfe 3892",
    "U oth 90",
    "U sas 282",
    "V afr 16",
    "V amr 14",
    "V asj 47",
    "V fin 273",
    "V nfe 869",
    "V oth 15",
    "W afr 15",
    "W amr 6",
    "W asj 23",
    "W fin 214",
    "W mid 1",
    "W nfe 508",
    "W oth 14",
    "W sas 38",
    "X afr 11",
    "X ami 25",
    "X amr 18",
    "X asj 9",
    "X fin 59",
    "X mid 1",
    "X nfe 405",
    "X oth 7",
    "X sas 11",
    "Y eas 11",
    "Y nfe 1",
    "Z eas 28",
    "Z fin 45",
    "Z nfe 11",
    "Z oth 1",
    "Z sas 4"
  ]
}
