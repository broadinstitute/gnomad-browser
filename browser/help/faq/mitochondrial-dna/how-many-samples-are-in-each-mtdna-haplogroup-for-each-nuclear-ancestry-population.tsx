import React, { useEffect, useState } from 'react'
import styled from 'styled-components'

import { BaseTable, Select } from '@gnomad/ui'

import haplogroupAndAncestryData, {
  AncestryName,
  codeToAncestryName,
  MtdnaHaplogroupSampleCount,
} from './mtdnaHaplogroupPerNuclearAncestryPopulationData'

import { haplogroupLabels } from '../../../src/MitochondrialVariantPage/MitochondrialVariantPage'

const HaplogroupAndAncestryBaseTable = styled(BaseTable)`
  margin-top: 1em;
  margin-bottom: 1em;

  td {
    text-align: right;
  }
`

const HaplogroupOrAncestrySelector = styled(Select)`
  margin: 1em 1em 0 0;
`

const HaplogroupAndAncestryFilterTable = () => {
  const [haplogroupSelected, setHaplogroupSelected] = useState('All')
  const [ancestrySelected, setAncestrySelected] = useState('All')
  const [filteredData, setFilteredData] = useState<MtdnaHaplogroupSampleCount[]>([])

  const ancestryOptions: AncestryName[] = [
    'African/African American',
    'Amish',
    'Latino/Admixed American',
    'Ashkenazi Jewish',
    'East Asian',
    'European (Finnish)',
    'Middle Eastern',
    'European (non-Finnish)',
    'South Asian',
    'Other',
  ]

  useEffect(() => {
    const newFilteredData = haplogroupAndAncestryData.filter((row) => {
      return (
        (row.haplogroup === haplogroupSelected || haplogroupSelected === 'All') &&
        (codeToAncestryName[row.ancestry] === ancestrySelected || ancestrySelected === 'All')
      )
    })
    setFilteredData(newFilteredData)
  }, [haplogroupSelected, ancestrySelected])

  return (
    <>
      <HaplogroupOrAncestrySelector
        id="haplogroup-selected"
        value={haplogroupSelected}
        onChange={(e) => {
          setHaplogroupSelected(e.target.value)
        }}
      >
        <optgroup label="Haplogroup">
          <option value="All" key="All">
            All Haplogroups
          </option>
          {haplogroupLabels.map((haplogroup) => (
            <option value={haplogroup} key={haplogroup}>
              {haplogroup}
            </option>
          ))}
        </optgroup>
      </HaplogroupOrAncestrySelector>

      <HaplogroupOrAncestrySelector
        id="ancestry-selected"
        value={ancestrySelected}
        onChange={(e) => {
          setAncestrySelected(e.target.value)
        }}
      >
        <optgroup label="Ancestry">
          <option value="All" key="All">
            All Ancestries
          </option>
          {ancestryOptions.map((ancestry) => (
            <option value={ancestry} key={ancestry}>
              {ancestry}
            </option>
          ))}
        </optgroup>
      </HaplogroupOrAncestrySelector>

      <HaplogroupAndAncestryBaseTable>
        <thead>
          <tr>
            <th scope="col">Haplogroup</th>
            <th scope="col">Population</th>
            <th scope="col">Number of Samples</th>
          </tr>
        </thead>

        <tbody>
          {filteredData.map((row) => {
            return (
              <tr key={row.haplogroup + row.ancestry}>
                <th scope="row">{row.haplogroup}</th>
                <th>{codeToAncestryName[row.ancestry]}</th>
                <td>{row.n}</td>
              </tr>
            )
          })}
        </tbody>

        <tfoot>
          <tr>
            <th scope="row">Haplogroup: {haplogroupSelected}</th>
            <th>Ancestry: {ancestrySelected}</th>
            <td>
              Samples:{' '}
              {filteredData.reduce((acc, row) => {
                return acc + row.n
              }, 0)}
            </td>
          </tr>
        </tfoot>
      </HaplogroupAndAncestryBaseTable>
    </>
  )
}

export const question =
  'How many samples are in each mtDNA haplogroup, for each nuclear ancestry population?'

export const renderAnswer = () => <HaplogroupAndAncestryFilterTable />
