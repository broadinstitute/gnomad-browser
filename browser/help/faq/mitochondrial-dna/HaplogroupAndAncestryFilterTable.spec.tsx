import { test, expect } from '@jest/globals'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { renderAnswer } from './how-many-samples-are-in-each-mtdna-haplogroup-for-each-nuclear-ancestry-population'

test('has no unexpected changes', () => {
  const tree = renderer.create(renderAnswer())
  expect(tree).toMatchSnapshot()
})
