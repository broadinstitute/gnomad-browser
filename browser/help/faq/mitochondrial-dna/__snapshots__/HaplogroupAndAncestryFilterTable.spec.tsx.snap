// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has no unexpected changes 1`] = `
[
  .c0 {
  height: calc(2em + 2px);
  padding: 0.375em 1.5em 0.375em 0.75em;
  border: 1px solid #6c757d;
  border-radius: 0.25em;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-image: url('data:image/gif;base64,R0lGODlhFQAEAIAAACMtMP///yH5BAEAAAEALAAAAAAVAAQAAAINjB+gC+jP2ptn0WskLQA7');
  background-position: center right;
  background-repeat: no-repeat;
  font-size: 1em;
  line-height: 1.25;
  margin: 1em 1em 0 0;
}

.c0:focus {
  outline: none;
  border-color: rgb(70,130,180);
  box-shadow: 0 0 0 0.2em rgba(70,130,180,0.5);
}

<select
    className="c0"
    id="haplogroup-selected"
    onChange={[Function]}
    value="All"
  >
    <optgroup
      label="Haplogroup"
    >
      <option
        value="All"
      >
        All Haplogroups
      </option>
      <option
        value="A"
      >
        A
      </option>
      <option
        value="B"
      >
        B
      </option>
      <option
        value="C"
      >
        C
      </option>
      <option
        value="D"
      >
        D
      </option>
      <option
        value="E"
      >
        E
      </option>
      <option
        value="F"
      >
        F
      </option>
      <option
        value="G"
      >
        G
      </option>
      <option
        value="H"
      >
        H
      </option>
      <option
        value="HV"
      >
        HV
      </option>
      <option
        value="I"
      >
        I
      </option>
      <option
        value="J"
      >
        J
      </option>
      <option
        value="K"
      >
        K
      </option>
      <option
        value="L0"
      >
        L0
      </option>
      <option
        value="L1"
      >
        L1
      </option>
      <option
        value="L2"
      >
        L2
      </option>
      <option
        value="L3"
      >
        L3
      </option>
      <option
        value="L4"
      >
        L4
      </option>
      <option
        value="L5"
      >
        L5
      </option>
      <option
        value="M"
      >
        M
      </option>
      <option
        value="N"
      >
        N
      </option>
      <option
        value="P"
      >
        P
      </option>
      <option
        value="R"
      >
        R
      </option>
      <option
        value="T"
      >
        T
      </option>
      <option
        value="U"
      >
        U
      </option>
      <option
        value="V"
      >
        V
      </option>
      <option
        value="W"
      >
        W
      </option>
      <option
        value="X"
      >
        X
      </option>
      <option
        value="Y"
      >
        Y
      </option>
      <option
        value="Z"
      >
        Z
      </option>
    </optgroup>
  </select>,
  .c0 {
  height: calc(2em + 2px);
  padding: 0.375em 1.5em 0.375em 0.75em;
  border: 1px solid #6c757d;
  border-radius: 0.25em;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-image: url('data:image/gif;base64,R0lGODlhFQAEAIAAACMtMP///yH5BAEAAAEALAAAAAAVAAQAAAINjB+gC+jP2ptn0WskLQA7');
  background-position: center right;
  background-repeat: no-repeat;
  font-size: 1em;
  line-height: 1.25;
  margin: 1em 1em 0 0;
}

.c0:focus {
  outline: none;
  border-color: rgb(70,130,180);
  box-shadow: 0 0 0 0.2em rgba(70,130,180,0.5);
}

<select
    className="c0"
    id="ancestry-selected"
    onChange={[Function]}
    value="All"
  >
    <optgroup
      label="Ancestry"
    >
      <option
        value="All"
      >
        All Ancestries
      </option>
      <option
        value="African/African American"
      >
        African/African American
      </option>
      <option
        value="Amish"
      >
        Amish
      </option>
      <option
        value="Latino/Admixed American"
      >
        Latino/Admixed American
      </option>
      <option
        value="Ashkenazi Jewish"
      >
        Ashkenazi Jewish
      </option>
      <option
        value="East Asian"
      >
        East Asian
      </option>
      <option
        value="European (Finnish)"
      >
        European (Finnish)
      </option>
      <option
        value="Middle Eastern"
      >
        Middle Eastern
      </option>
      <option
        value="European (non-Finnish)"
      >
        European (non-Finnish)
      </option>
      <option
        value="South Asian"
      >
        South Asian
      </option>
      <option
        value="Other"
      >
        Other
      </option>
    </optgroup>
  </select>,
  .c0 {
  border-collapse: collapse;
  border-spacing: 0;
  margin-top: 1em;
  margin-bottom: 1em;
}

.c0 td,
.c0 th {
  padding: 0.5em 10px 0.5em 0;
  text-align: left;
}

.c0 thead td,
.c0 thead th {
  border-bottom: 1px solid #000;
  background-position: center right;
  background-repeat: no-repeat;
}

.c0 thead td[aria-sort='ascending'],
.c0 thead th[aria-sort='ascending'] {
  background-image: url('data:image/gif;base64,R0lGODlhFQAEAIAAACMtMP///yH5BAEAAAEALAAAAAAVAAQAAAINjI8Bya2wnINUMopZAQA7');
}

.c0 thead td[aria-sort='descending'],
.c0 thead th[aria-sort='descending'] {
  background-image: url('data:image/gif;base64,R0lGODlhFQAEAIAAACMtMP///yH5BAEAAAEALAAAAAAVAAQAAAINjB+gC+jP2ptn0WskLQA7');
}

.c0 thead td button,
.c0 thead th button {
  padding: 0;
  border: none;
  background: none;
  color: inherit;
  cursor: pointer;
  font: inherit;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 tbody td,
.c0 tbody th {
  border-bottom: 1px solid #ccc;
  font-weight: normal;
}

.c0 tfoot td,
.c0 tfoot th {
  border-top: 1px solid #ccc;
  font-weight: bold;
}

.c0 td {
  text-align: right;
}

<table
    className="c0"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Haplogroup
        </th>
        <th
          scope="col"
        >
          Population
        </th>
        <th
          scope="col"
        >
          Number of Samples
        </th>
      </tr>
    </thead>
    <tbody />
    <tfoot>
      <tr>
        <th
          scope="row"
        >
          Haplogroup: 
          All
        </th>
        <th>
          Ancestry: 
          All
        </th>
        <td>
          Samples:
           
          0
        </td>
      </tr>
    </tfoot>
  </table>,
]
`;
